src/FakeItEasy/ExceptionThrowerConfigurationExtensions.cs

```diff
@@ -130,6 +134,11 @@ public static IAfterCallSpecifiedConfiguration Throws<T1>(this IExceptionThrower
                 });
         }
 
+#if DISABLED
+
+        // This method already has a generic parameter, and it can't be inferred; if we add another, we make it unusable,
+        // especially since the new one is something unintelligible like IAnyCallConfigurationWithReturnTypeSpecificied<Foo>...
+

```

Comment:
I've always seen these extension methods as coding convenience, not necessity. Could we make this not be an extension method? e.g., on `RuleBuilder` (and other leaf classes) have a method like

```c#
public IAfterCallSpecifiedConfiguration<IVoidConfiguration> Throws<T>() where T : Exception, new()
{
    return this.Throws(() => new T());
}
```

If I add this, and tart up `IVoidArgumentValidationConfiguration` (possibly not the best place, but it was expedient), then this passes, where `Bar` returns `void`, if I can say that:

```c#
[Scenario]
public static void Throws(
    IFoo fake,
    Exception exception)
{
    "Given a fake"
        .x(() => fake = A.Fake<IFoo>());

    "And I configure a method to throw"
        .x(() => A.CallTo(() => fake.Bar()).Throws<ArgumentOutOfRangeException>());

    "When I call the method"
        .x(() => exception = Record.Exception(() => fake.Bar()));

    "Then it throws"
        .x(() => exception.Should().BeAnExceptionOfType<ArgumentOutOfRangeException>());
}
```